primitives/arithmetic/src/fixed_point.rs

```diff
@@ -536,10 +536,10 @@ macro_rules! implement_fixed {
 			}
 		}
 
-		impl<P: PerThing> From<P> for $name {
+		impl<P: PerThing> From<P> for $name where P::Inner: FixedPointOperand {
 			fn from(p: P) -> Self {
-				let accuracy = P::ACCURACY.saturated_into();
-				let value = p.deconstruct().saturated_into();
+				let accuracy = P::ACCURACY;
+				let value = p.deconstruct();
```

Comment:
I'm surprised the former version compiled in the first place, what type does it sature into ? it seems u32 u64, u128 could all be picked no ?

Anyway it can be considered a bugfix to me. If `PerThing::Inner` is U256 then this would saturate way too much.

Reply:
ah :-/ thanks

then it is a bugfix to me. As if the compiler use to saturate into u32 for Perquintill the result is pretty bad.